src/sqlfluff/dialects/dialect_mariadb_keywords.py

```diff
@@ -0,0 +1,706 @@
+INVISIBLE
+IO
+IO_THREAD
+IPC
+ISOLATION
+ISOPEN
+ISSUER
+INVOKER
+JSON
+JSON_TABLE
+KEY_BLOCK_SIZE
+LANGUAGE
+LAST
+LAST_VALUE
+LASTVAL
+LEAVES
+LESS
+LEVEL
+LIST
+LOCAL
+LOCKED
+LOCKS
+LOGFILE
+LOGS
+MASTER
+MASTER_CONNECT_RETRY
+MASTER_DELAY
+MASTER_GTID_POS
+MASTER_HOST
+MASTER_LOG_FILE
+MASTER_LOG_POS
+MASTER_PASSWORD
+MASTER_PORT
+MASTER_SERVER_ID
+MASTER_SSL
+MASTER_SSL_CA
+MASTER_SSL_CAPATH
+MASTER_SSL_CERT
+MASTER_SSL_CIPHER
+MASTER_SSL_CRL
+MASTER_SSL_CRLPATH
+MASTER_SSL_KEY
+MASTER_USER
+MASTER_USE_GTID
+MASTER_DEMOTE_TO_SLAVE
+MAX_CONNECTIONS_PER_HOUR
+MAX_QUERIES_PER_HOUR
+MAX_ROWS
+MAX_SIZE
+MAX_STATEMENT_TIME
+MAX_UPDATES_PER_HOUR
+MAX_USER_CONNECTIONS
+MEDIUM
+MEMORY
+MERGE
+MESSAGE_TEXT
+MICROSECOND
+MIGRATE
+MINUS
+MINUTE
+MINVALUE
+MIN_ROWS
+MODE
+MODIFY
+MONITOR
+MONTH
+MUTEX
+MYSQL
+MYSQL_ERRNO
+NAME
+NAMES
+NATIONAL
+NCHAR
+NESTED
+NEVER
+NEXT
+NEXTVAL
+NO
+NOMAXVALUE
+NOMINVALUE
+NOCACHE
+NOCYCLE
+NO_WAIT
+NOWAIT
+NODEGROUP
+NONE
+NOTFOUND
+NUMBER
+NVARCHAR
+OF
+OLD_PASSWORD
+ONE
+ONLINE
+ONLY
+OPEN
+OPTIONS
+ORDINALITY
+OTHERS
+OVERLAPS
+OWNER
+PACKAGE
+PACK_KEYS
+PAGE
+PARSER
+PATH
+PERIOD
+PARTIAL
+PARTITIONING
+PARTITIONS
+PASSWORD
+PERSISTENT
+PHASE
+PLUGIN
+PLUGINS
+PORT
+PORTION
+PRECEDES
+PRECEDING
+PREPARE
+PRESERVE
+PREV
+PREVIOUS
+PRIVILEGES
+PROCESS
+PROCESSLIST
+PROFILE
+PROFILES
+PROXY
+QUARTER
+QUERY
+QUERY_RESPONSE_TIME
+QUICK
+RAISE
+RAW
+READ_ONLY
+REBUILD
+RECOVER
+REDO_BUFFER_SIZE
+REDOFILE
+REDUNDANT
+RELAY
+RELAYLOG
+RELAY_LOG_FILE
+RELAY_LOG_POS
+RELAY_THREAD
+RELOAD
+REMOVE
+REORGANIZE
+REPAIR
+REPEATABLE
+REPLAY
+REPLICA
+REPLICAS
+REPLICA_POS
+REPLICATION
+RESET
+RESTART
+RESTORE
+RESUME
+RETURNED_SQLSTATE
+RETURNS
+REUSE
+REVERSE
+ROLE
+ROLLBACK
+ROLLUP
+ROUTINE
+ROW
+ROWCOUNT
+ROWNUM
+ROWTYPE
+ROW_COUNT
+ROW_FORMAT
+RTREE
+SAVEPOINT
+SCHEDULE
+SCHEMA_NAME
+SECOND
+SECURITY
+SEQUENCE
+SERIAL
+SERIALIZABLE
+SESSION
+SERVER
+SETVAL
+SHARE
+SHUTDOWN
+SIGNED
+SIMPLE
+SKIP
+SLAVE
+SLAVES
+SLAVE_POS
+SNAPSHOT
+SOCKET
+SOFT
+SOME
+SONAME
+SOUNDS
+SOURCE
+STAGE
+STORED
+SQL_AFTER_GTIDS
+SQL_BEFORE_GTIDS
+SQL_BUFFER_RESULT
+SQL_CACHE
+SQL_NO_CACHE
+SQL_THREAD
+SQL_TSI_SECOND
+SQL_TSI_MINUTE
+SQL_TSI_HOUR
+SQL_TSI_DAY
+SQL_TSI_WEEK
+SQL_TSI_MONTH
+SQL_TSI_QUARTER
+SQL_TSI_YEAR
+START
+STARTS
+STATEMENT
+STATUS
+STOP
+STORAGE
+STRING
+SUBCLASS_ORIGIN
+SUBJECT
+SUBPARTITION
+SUBPARTITIONS
+SUPER
+SUSPEND
+SWAPS
+SWITCHES
+SYSDATE
+SYSTEM
+SYSTEM_TIME
+TABLE_NAME
+TABLES
+TABLESPACE
+TABLE_CHECKSUM
+TABLE_STATISTICS
+TEMPORARY
+TEMPTABLE
+TEXT
+THAN
+TIES
+TIME
+TIMESTAMP
+TIMESTAMPADD
+TIMESTAMPDIFF
+TRANSACTION
+TRANSACTIONAL
+THREADS
+TRIGGERS
+TRUNCATE
+TYPE
+UNBOUNDED
+UNCOMMITTED
+UNDEFINED
+UNDO_BUFFER_SIZE
+UNDOFILE
+UNICODE
+UNKNOWN
+UNINSTALL
+UNTIL
+UPGRADE
+USER
+USER_RESOURCES
+USER_STATISTICS
+USER_VARIABLES
+USE_FRM
+VALIDATION
+VALUE
+VARCHAR2
+VARIABLES
+VIA
+VIEW
+VIRTUAL
+VISIBLE
+VERSIONING
+WAIT
+WARNINGS
+WEEK
+WEIGHT_STRING
+WITHIN
+WITHOUT
+WORK
+WRAPPER
+X509
+XA
+XML
+YEAR
+"""
```

Comment:
The `mysql` dialect has a couple extra "keywords" at the end: https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_mysql_keywords.py#L784

Are you sure those aren't needed, given that this extends the mysql dialect?

Reply:
They weren't listed in the mariadb repo or docs as keywords, so I don't know if they're supported or not. Might be worth exploring, as I've never seen or used some of these (e.g., `Zone` or `XID`)